src/__tests__/core/index.test.js

```diff
@@ -47,19 +49,199 @@ describe('setup', () => {
     const { mock } = mockInit;
     expect(mock.calls.length).toBe(1);
     const model = mock.calls[0][1].toJS();
     expect(model).toMatchSnapshot();
   });
+
+  describe('clientBaseUrl', () => {
+    it('should default to the specified domain', () => {
+      const { mock } = mockInit;
+
+      setup(
+        'id',
+        'clientID',
+        'my-tenant.us.auth0.com',
+        {},
+        'hookRunner',
+        'emitEventFn',
+        'handleEventFn'
+      );
+
+      expect(mock.calls.length).toBe(1);
+
+      const model = mock.calls[0][1].toJS();
+      expect(model.clientBaseUrl).toBe('https://my-tenant.us.auth0.com');
+    });
+
+    it('should use the clientBaseUrl option if given', () => {
+      const { mock } = mockInit;
+
+      setup(
+        'id',
+        'clientID',
+        'my-tenant.us.auth0.com',
+        {
+          clientBaseUrl: 'https://client-base-url.example.com',
+          configurationBaseUrl: 'https://config-base-url.example.com',
+          assetsUrl: 'https://assets-url.example.com'
+        },
+        'hookRunner',
+        'emitEventFn',
+        'handleEventFn'
+      );
+
+      expect(mock.calls.length).toBe(1);
+
+      const model = mock.calls[0][1].toJS();
+      expect(model.clientBaseUrl).toBe('https://client-base-url.example.com');
+    });
+
+    it('should use configurationBaseUrl if given', () => {
+      const { mock } = mockInit;
+
+      setup(
+        'id',
+        'clientID',
+        'my-tenant.us.auth0.com',
+        {
+          configurationBaseUrl: 'https://config-base-url.example.com',
+          assetsUrl: 'https://assets-url.example.com'
+        },
+        'hookRunner',
+        'emitEventFn',
+        'handleEventFn'
+      );
+
+      expect(mock.calls.length).toBe(1);
+
+      const model = mock.calls[0][1].toJS();
+      expect(model.clientBaseUrl).toBe('https://config-base-url.example.com');
+    });
+
+    it('should use assetsUrl if given', () => {
+      const { mock } = mockInit;
+
+      setup(
+        'id',
+        'clientID',
+        'my-tenant.us.auth0.com',
+        {
+          assetsUrl: 'https://assets-url.example.com'
+        },
+        'hookRunner',
+        'emitEventFn',
+        'handleEventFn'
+      );
+
+      expect(mock.calls.length).toBe(1);
+
+      const model = mock.calls[0][1].toJS();
+      expect(model.clientBaseUrl).toBe('https://assets-url.example.com');
+    });
+  });
+
+  describe('tenantBaseUrl', () => {
+    it('tenantBaseUrl should default to domain URL when using auth0.com', () => {
+      const { mock } = mockInit;
+
+      setup(
+        'id',
+        'clientID',
+        'my-tenant.us.auth0.com',
+        {
+          __useTenantInfo: true
+        },
+        'hookRunner',
+        'emitEventFn',
+        'handleEventFn'
+      );
+
+      expect(mock.calls.length).toBe(1);
+
+      const model = mock.calls[0][1].toJS();
+      expect(model.tenantBaseUrl).toBe('https://my-tenant.us.auth0.com/tenants/v1/my-tenant.js');
```

Comment:
"tenant base URL" is then compared against a URL that also has a complete path to a file (tenant.js), so might not really represent a "base URL"

Reply:
I agree, the naming isn't the best. However, the name exists from way before these changes,